libs/sdk/tools/src/generators/update-dependencies/generator.ts

```diff
@@ -149,27 +154,39 @@ function updateTestingImplicitDependencies(tree: Tree): void {
       projectConfig.targets ??= {};
       projectConfig.targets['build'] = {
         command: `echo ' 🏗️  build ${projectConfig.name}'`,
-        dependsOn: [
-          {
-            projects: [projectConfig.name.replace('-testing', '')],
-            target: 'build',
-          },
-        ],
+        inputs: ['buildInputs'],
+        dependsOn: [],
```

Comment:
Avoid circular dependency, e.g., `core-testing:build` depending on `core:build`, by overriding `nx.json` file `targetDefaults` for testing projects.